docs/docs/workflows/terraform-provider.md

```diff
@@ -0,0 +1,195 @@
+
+  <tabItem value="gcp" label="GCP">
+
+  ```
+  module "gcp-constellation" {
+    source = "https://github.com/edgelesssys/constellation/releases/download/<version>/terraform-module.zip//terraform-module/gcp-constellation" // replace <version> with a Constellation version, e.g., v2.13.0
+    name    = "constell"
+    project = "constell-proj" // replace with your project id
+    service_account_id = "constid"
+    zone    = "europe-west2-a"
+    node_groups = {
+      control_plane_default = {
+        role          = "control-plane"
+        zone          = "europe-west2-a"
+        instance_type = "n2d-standard-4"
+        disk_size     = 30
+        disk_type     = "pd-ssd"
+        initial_count = 3
+      },
+      worker_default = {
+        role          = "worker"
+        zone          = "europe-west2-a"
+        instance_type = "n2d-standard-4"
+        disk_size     = 30
+        disk_type     = "pd-ssd"
+        initial_count = 2
+      }
+    }
+  }
+  ```
+
+  </tabItem>
+  </tabs>
+
+1. Initialize and apply the file.
+  ```bash
+  terraform init
+  terraform apply
+  ```
+
+## Custom setup
+If you need custom infrastructure, you can download the Terraform module from the Constellation [GitHub releases](https://github.com/edgelesssys/constellation/releases) and modify it.
+The module contains:
+- `{csp}`: cluster infrastructure resources for the respective cloud provider
+- `iam/{csp}`: contains the IAM resources used within the cluster
+
+For cluster upgrades, make sure to check the Constellation release notes for potential breaking changes in the infrastructure setup.
+
+## Cluster upgrades
+:::tip
+For general information on cluster upgrades, see [Upgrade your cluster](./upgrade.md).
+:::
+
+First update the version of the Constellation Terraform provider. If you explicitly set versions (e.g. `image_version` or `constellation_microservice_version`), make sure to update them. Refer to [version support](https://github.com/edgelesssys/constellation/blob/main/dev-docs/workflows/versions-support.md) for more information on the version support policy.
+Regarding the infrastructure / IAM modules, using a [remote address as module source](https://developer.hashicorp.com/terraform/language/modules/sources#fetching-archives-over-http) as shown in [Quick setup](#quick-setup) is recommended because it simplifies the upgrade process. For [local paths as module source](https://developer.hashicorp.com/terraform/language/modules/sources#local-paths), you need to replace the local files with those from the [Constellation release](https://github.com/edgelesssys/constellation/releases). Also look for potential breaking changes in the infrastructure setup in the release notes.
+
+The steps for applying the upgrade are as follows:
+
+1. Update the `<version>` variable inside the `source` field of the infra / IAM module. If explicitly set, update the `constellation_microservice_version` and `image` fields of the `constellation_cluster` resource.
```

Comment:
```suggestion
1. If using the reference infrastructure / IAM modules, update the URL in the `source` field of the modules to the `terraform-module.zip` artifact URL for the upgrade target version found on the [release page](https://github.com/edgelesssys/constellation/releases/latest). If using a custom infrastructure configuration, see the changes made in the reference modules since the upgrade's origin version and adjust your infrastructure configuration accordingly. If explicitly set, update the `constellation_microservice_version` and `image_version` fields of the Constellation Terraform provider's resources and data sources.
```

nit: Explaining in a little more detail here.

Also, the `<version>` variable is only there when initially copying the reference configuration from above, but not when upgrading a cluster.